### ISSUE TITLE
In UWP applications, if a link contains visible text as the name, it is not announced by NVDA

### ISSUE THREAD

**Issue by MarcoZehe** [nvaccess/nvda#9454]
### Steps to reproduce:

Found while using the [Unigram](https://www.microsoft.com/store/apps/9n97zckpd60q) Telegram client, which is an UWP application.

1. Set up a [Telegram](https://www.telegram.org) account if you don't have one.
2. Download Unigram from the above Store link and connect it to your account.
3. Create a conversation, and ensure that you have at least one reply in there:
   1. Focus the message you want to reply to and hit Context Menu key.
   2. Select Reply, then type your reply.
4. In the output list, focus your reply you just sent.
5. Tab once.

### Actual behavior:

NVDA will speak the group info for your message, and the word "Link", but not what the link's name is.

### Expected behavior:

It should speak the link's name, which is the text of the message this is a reply to. The text is a child of the link, done as a Textblock control, in accordance with [this information](https://docs.microsoft.com/en-us/windows/uwp/design/accessibility/basic-accessibility-information#name_from_inner_text) given by Microsoft's documentation. It appears that some Edge code introduced in 2016 special cased those exact controls like links and buttons, as well as some others, to no longer retrieve the name from text if I understand the references in [this code block](https://github.com/nvaccess/nvda/blame/master/source/NVDAObjects/UIA/__init__.py#L394) correctly.

#### NVDA installed/portable/running from source:

Installed.

#### NVDA version:

2019.1

#### Windows version:

10.0.1809 (Build 17763,379).

#### Name and version of other software in use when reproducing the issue:

Unigram latest version from Microsoft Store.

#### Other information about your system:

Not applicable.

#### Does the issue still occur after restarting your PC?

Yes.

#### Have you tried any other versions of NVDA?

Both 2018.4 as well as latest master, both also show the issue.

**Comment by MarcoZehe** [comment]
Additional info: Narrator does speak the link properly.

**Comment by Adriani90** [comment]
@MarcoZehe any updates on this? Is it still reproducible with last Alpha version of NVDA?

**Comment by MarcoZehe** [comment]
Nope, still an issue.

**Comment by Adriani90** [comment]
This might be related to #16415.
